python_tools/pixmob_conversion_funcs.py

```diff
@@ -1,61 +1,73 @@
-import copy
-
-# This file contains functions used for converting between different representations of the PixMob data packets,
-# including "logic level" (time between IR light high/low switches), binary (high/low IR light value at each unit of 700
-# microseconds), hexadecimal, and the string representations parsed by the Arduino to send the IR signals.
-logic_level_to_data_num = {
-    1: 700,
-    2: 1400,
-    3: 2100,
-    4: 2800,
-    5: 3500,
-    #6: 4200,
-
-    9: 6300,
-}
-def to_bits(level_list):
-    curr_state = 1
+import itertools
+import python_tools.config as cfg
+
+
+# This file contains functions used for converting between different representations of the PixMob data
+# packets, including:
+#
+# - "run length" (time between IR light high/low switches), measured in microseconds, and also in pulses(could be called
+#       bit widths or chunks)
+# - "bits" (high/low IR light value at each unit of 694.44 microseconds)
+# - hexadecimal
+# - string representations parsed by the Arduino to send the IR signals.
+
+def bits_to_hex(bit_list):
+    # Example: [1, 1, 1, 1, 0, 0, 0, 0] -> 0xf0
+    return hex(int(("".join([str(i) for i in bit_list])), 2))
+
+
+def bits_to_run_lengths_pulses(bit_list):
+    # convert from a list of 1s and 0s to run length by number of pulses
+    # Example: [1, 1, 1, 0, 0, 0, 0, 1] -> [3, 4, 1]
+    # TODO: Throw an exception if the bit list isn't just ones and zeroes?
+    # TODO: Combine this with the other bit to run length function and let this one be the case where pulse length is 1
+    run_lengths = []
+    # groupby returns groups of adjacent matching things
+    for _, group in itertools.groupby(bit_list):
+        run_lengths.append(sum(1 for _ in group))
+    return run_lengths
+
+
+def bits_to_run_lengths_microseconds(bit_list, pulse_length=cfg.PULSE_LENGTH):
```

Comment:
Thanks for making it possible to override `pulse_length` from the config file at runtime if needed.